### ISSUE TITLE
1.2 C

### ISSUE THREAD

**Issue by sonali-sr** [rebeccajohnson88/PPOL564_slides_activities#51]
### 1.2 named entity recognition (3 points)

C. You want to extract the possible sentence lengths the CEO is facing; pull out the named entities with (1) the label DATE and (2) that contain the word year or years. Print these named entities.

**_Hint:_**
You may want to use the re module for the second part.

<img width="516" alt="1 2C" src="https://user-images.githubusercontent.com/90288860/197810415-097766ff-03c0-46d0-b990-ed1cdb2a20e8.PNG">

**Comment by JiaqinWu** [comment]
After subsetting after using this code, I get only three variables. Here's my 

> code:pharma = str(doj[doj.id == '17-1204'].contents)
spacy_pharma = nlp(pharma)
for i in spacy_pharma.ents:
    print("Entity: " + i.text + "; NER tag: " + i.label_)

**Comment by qinip** [comment]
Hi, my result of `npl(pharma).ents` doesn't contain "20 years" or "no greater than five years," what could I have done wrong? Thanks!

**Comment by JiaqinWu** [comment]
> Hi, my result of `npl(pharma).ents` doesn't contain "20 years" or "no greater than five years," what could I have done wrong? Thanks!

me too...

**Comment by rebeccajohnson88** [comment]
@JiaqinWu @qinip my guess is that this is a spaCy versioning difference, in which case:

- we'll give credit for the outputs produced by EITHER version and
- it's fine that you don't match the above exactly

can you run these commands and screenshot the result so we can compare versions? i can then update mine (if yours is more current) and post the output from that

```
!pip install session_info
import session_info
session_info.show()
```


![image](https://user-images.githubusercontent.com/21369316/197876908-aceb05eb-d4c2-4f83-857b-2d59bfcccdba.png)

**Comment by rebeccajohnson88** [comment]
got it! i'll upgrade or have a TA upgrade and post the printout from running with the newer version

for now, you're fine and i'd move onto the other problems

**Comment by qinip** [comment]
<img width="699" alt="Screenshot 2022-10-25 at 4 40 08 PM" src="https://user-images.githubusercontent.com/5789293/197877545-4dc4e859-d9f1-4a9b-91d3-5d2262286cd3.png">

**Comment by jswsean** [comment]
> Hi, my result of `npl(pharma).ents` doesn't contain "20 years" or "no greater than five years," what could I have done wrong? Thanks!

I also got `['last year', 'three years', 'three years']` as my output. My package versions are as follows:

![image](https://user-images.githubusercontent.com/54017275/197906867-cff6e785-b891-4cac-a3b2-3b31e8cb6bd1.png)

**Comment by rebeccajohnson88** [comment]
@jswsean yep i think that's output from a newer version - @sonali-sr can share when she produces output from that newer version. in meantime, fine to proceed and we'll give credit for both outputs since some OS may not be compatible with 3.9- thanks!

**Comment by sonali-sr** [comment]
Hello everyone, attached is the revised output for this question. Some of you all may have the output produced earlier - and as Prof. Johnson mentioned, both are fine. 

<img width="491" alt="Capture" src="https://user-images.githubusercontent.com/90288860/198185979-49f1f91c-9839-4b6e-b59e-d04c5aad0591.PNG">

**Comment by bhollan** [comment]
I'm on an intermediary version of spacy and am getting something unique from either of the "both" mentioned.

![image](https://user-images.githubusercontent.com/8443608/199647982-acb957af-a15f-444b-85ec-f827693dbda8.png)


I'm getting this:
```
{'20 years', 'last year', 'three years'}
```

**Comment by bhollan** [comment]
Now I forced the install of `3.4.2`.

![image](https://user-images.githubusercontent.com/8443608/199648422-6743d042-edec-4327-8988-e7a3278029d5.png)

But it's still giving me the same as before:

```
{'20 years', 'last year', 'three years'}
```

**Comment by rebeccajohnson88** [comment]
@bhollan yea the versioning stuff on this is puzzling; TAs will be grading based on code rather than output as a result so yours is fine- thanks!
